docs/upgrading.md

```diff
@@ -0,0 +1,28 @@
+# Upgrading
+
+## Upgrading from v2 to v3.0.0
+
+Starting from `external-dns-app` version `3.0.0`, there have been modifications to the values.yaml schema. These changes were introduced in version `2.28.0`, and the previous format has been completely removed in version 3.0.0. Therefore, it's important to take note of these updates while upgrading.
+
+### Values
+
+- `aws.access` has been removed. Check the [External DNS with AWS Route 53 and static credentials](https://docs.giantswarm.io/advanced/external-dns/aws-route53-static-creds) guide for more details.
+- `aws.accountID` has been removed. Use `serviceAccount.annotations` to inject the `eks.amazonaws.com/role-arn` annotation containing the proper IAM role ARN into the Service Account.
+- `aws.baseDomain` has been removed. Use `domainFilters` instead.
+- `aws.batchChangeSize`, `aws.batchChangeInterval`, `aws.preferCNAME` and `aws.zonesCacheDuration` have been removed. Use the arguments `--aws-batch-change-size=N`, `--aws-batch-change-interval=SECONDS`, `--aws-prefer-cname` and `--aws-zones-cache-duration=SECONDS` in `extraArgs` if desired.
+- `aws.irsa` is no longer needed, as all our clusters now support IRSA.
+- `aws.region` has been removed. Set the region if needed injecting the AWS_DEFAULT_REGION variable using the `env` value.
+- Additional arguments must be configured by setting `extraArgs` instead of `externalDNS.extraArgs`
+- `externalDNS.annotationFilter` has been removed in favor of `annotationFilter`.
+- Use `secretConfiguration` in replacement of `externalDNS.aws_access_key_id` and `externalDNS.aws_secret_access_key`. Both values have been removed.
+- `externalDNS.domainFilterList` has been replaced with `domainFilters`
```

Comment:
It may be possible to forbid the `externalDNS` (if removed completely) key through the values schema (https://stackoverflow.com/a/70266255)

This would prevent installation with the deprecated values schema.

Reply:
that is possible, it might be a bit dangerous because the externalDNS value is injected via catalog. That would mean that old installations might break.